irohad/ametsuchi/impl/storage_impl.cpp

```diff
@@ -368,18 +385,23 @@ namespace iroha {
                 [&](expected::Value<std::shared_ptr<soci::connection_pool>>
                         &connection) {
                   soci::session sql(*connection.value);
-                  bool enable_prepared_transactions =
-                      preparedTransactionsAvailable(sql);
-                  storage = expected::makeValue(std::shared_ptr<StorageImpl>(
-                      new StorageImpl(block_store_dir,
-                                      options,
-                                      std::move(ctx.value.block_store),
-                                      connection.value,
-                                      factory,
-                                      converter,
-                                      perm_converter,
-                                      pool_size,
-                                      enable_prepared_transactions)));
+                  preparedTransactionsAvailable(sql).match(
```

Comment:
instead of match, consider using operator |

```C++
storage = preparedTransactionsAvailable(sql) |
                      [&](bool available) -> decltype(storage) {
                    return expected::makeValue(std::shared_ptr<StorageImpl>(
                        new StorageImpl(block_store_dir,
                                        options,
                                        std::move(ctx.value.block_store),
                                        connection.value,
                                        factory,
                                        converter,
                                        perm_converter,
                                        pool_size,
                                        available)));
                  };
```

Reply:
Instantiation can only fail if the exception is thrown. In that case both variants are equivalent in vehavior.